cloud/amazon/nat_gateway.py

```diff
@@ -0,0 +1,393 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+module: nat_gateway
+short_description: Create, delete and describe AWS Managed NAT Gateways.
+Requires Boto3.
+description:
+  - Creates AWS Managed NAT Gateways with option to provide EIP or
+    allocate new address.
+  - Deletes AWS Managed NAT Gateways with option to release attached EIP.
+  - Describe AWS Managed NAT Gateways, either all or with Filters
+  - This module does not support updates or tagging as it is a current
+    restriction with AWS.
+  - This module support check mode.
+version_added: "2.0"
+options:
+  describe_action:
+    description:
+      - specifies the action to be a get action, to describe existing
+        nat gateways
+    required: false
+  describe_filter:
+    description:
+      - Use in conjunction with describe_action to filter results
+      - See U(http://boto3.readthedocs.org/en/latest/reference/services/ec2.html#EC2.Client.describe_nat_gateways)
+        for possible filters. Only the Filter section is allowed in this module.
+    required: false
+  subnet_id:
+    description:
+      - Required when creating a NAT gateway.
+    required: false
+  eip_address:
+    description:
+      - An elastic IP already allocated in the AWS account to be used
+        by the NAT gateway. This address cannot already by attached to
+        another resource. If this option is not specified, an elastic IP
+        will be allocated and attached for you.
+    required: false
+  token:
+    description:
+      - A unique ASCII string up to 64 characters to identify the request.
+      - Required only for create actions.
+    required: false
+  state:
+    description:
+        - present to ensure resource is created.
+        - absent to remove resource
+    required: false
+    default: present
+    choices: [ "present", "absent"]
+  wait:
+    description:
+      - When specified, will wait for either available status for state present
+        or deleted for state absent. If release_eip is specified during state absent,
+        this will automatically be utilised.
+    required: false
+    default: no
+    choices: ["yes", "no"]
+  wait_timeout:
+    description:
+      - Used in conjunction with wait. Number of seconds to wait for status.
+        Recommended value of over 2 minutes.
+    required: false
+    default: 320
+  release_eip:
+    description:
+      - AWS Managed NAT Gateways do not release the elastic IP by default. Specify
+        this option if releasing eip is required.
+    required: false
+  nat_gateway_id:
+    description:
+      - The ID of the NAT gateway to be removed - used only for state absent
+    required: false
+author: Karen Cheng(@Etherdaemon)
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+- name: Get all existing nat gateways
+  nat_gateway:
+    describe_action: yes
```

Comment:
instead of making this an option, always return the data, as it seems you need it anyways.